src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.size = 0;
+        this.elements = new Object[size];
+    }
+
+    public ArrayList(Object[] elements, int size) {
+        this.size = size;
+        this.elements = elements;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
```

Comment:
Better have implementation here.
Call `add(T value)` inside a`dd(T value, int index)` when index == size

Reply:
rewrite both methods using ensureCapacity(); and  System.arraycopy(elements, index, elements, index + 1, size - index);